test/performance/io/format_fastq_benchmark.cpp

```diff
@@ -0,0 +1,185 @@
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+// written for the purpose of the Bch Softwarepraktikum, early 2019 FU Berlin
+// @author: Clemens Cords <clemenscords@fu-berlin.de>
+
+#include <fstream>
+#include <cstring>
+#include <cmath>
+
+#include <benchmark/benchmark.h>
+
+#include <seqan3/io/sequence_file/format_fastq.hpp>
+#include <seqan3/alphabet/nucleotide/dna5.hpp>
+#include <seqan3/alphabet/quality/phred42.hpp>
+#include <seqan3/io/sequence_file/input.hpp>
+#include <seqan3/range/view/to_char.hpp>
+
+using namespace seqan3;
+
+const std::size_t POWER = 5; //files will have |10^POWER| lines
+
+const auto DNA_SEQ = "AGCTAGCAGCGATCGCGATCGATCAGCGATCGAGGAATATAT"_dna4;
+const auto QUALITY = "IIIIIHIIIIIIIIIIIIIIIIIIIIIIHHGIIIIHHGIIIH"_phred42;
+
+// ============================================================================
+// generate dummy fastq file with 10^power identical 4-line entries
+// ============================================================================
+std::string generate_dummy_fastq_file(std::size_t power)
+{
+    std::string file{};
+
+    const std::string id{"@name"};
+    const std::string seq = DNA_SEQ | view::to_char;
+    const std::string qualid{"+"};
+    const std::string quality = QUALITY | view::to_char;
+
+    const unsigned int n_iterations = static_cast<unsigned int>(pow(10, power));
+
+    for (size_t i = 0; i < n_iterations; ++i)
+    {
+
+        file += id + '\n'
+                + seq + '\n'
+                + qualid + '\n'
+                + quality + '\n';
+    }
+
+    return file;
+}
+
+// ============================================================================
+// generate dummy fasta file with 2*10^power identical 2-line entries
+// ============================================================================
+std::string generate_dummy_fasta_file(std::size_t power)
+{
+    std::string file{};
+
+    const std::string input_id{">name"};
+    const std::string seq = DNA_SEQ | view::to_char;
+
+    const auto n_iterations = static_cast<unsigned int>(2*pow(10, power));
+
+    for (size_t i = 0; i < n_iterations; ++i)
+    {
+
+        file += input_id + '\n'
+              + seq + '\n';
+    }
+
+    return file;
+}
+
+// ============================================================================
+// try to write 4-line entry to stream as often as possible
+// ============================================================================
+void fastq_write(benchmark::State & state)
+{
+    std::ostringstream ostream;
+
+    sequence_file_format_fastq format;
+    sequence_file_output_options options{};
+
+    const std::string id{"@name"};
+    const std::string seq = DNA_SEQ | view::to_char;
```

Comment:
you can write the seq directly with the format! no need to convert to char here.